gtk4/src/subclass/layout_manager.rs

```diff
@@ -0,0 +1,294 @@
+use gtk_sys;
+
+use glib::subclass::object::ObjectImpl;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, Object};
+
+use libc::c_int;
+use std::mem;
+
+use LayoutChild;
+use LayoutManager;
+use Orientation;
+use Widget;
+
+pub trait LayoutManagerImpl: LayoutManagerImplExt + ObjectImpl {
+    fn allocate(
+        &self,
+        layout_manager: &Self::Type,
+        widget: &Widget,
+        width: i32,
+        height: i32,
+        baseline: i32,
+    ) {
+        self.parent_allocate(layout_manager, widget, width, height, baseline)
+    }
+    fn create_layout_child(
+        &self,
+        layout_manager: &Self::Type,
+        widget: &Widget,
+        for_child: &Widget,
```

Comment:
Can any of those widgets be `NULL`?

Reply:
From what I understood from the C API the widget calls this function from the layout manager by passing it a pointer of itself as `widget` & the child widget as `for_child` . So they can't be nullable